cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -408,11 +408,31 @@ def test_suspend(self, verify_vm_running, vm):
         vm.action.suspend()
         wait_for(lambda: vm.power_state == "suspended", num_sec=300, delay=5, fail_func=vm.reload)
 
+    def test_reset(self, verify_vm_running, vm):
+        assert "reset" in vm.action
+        vm.action.reset()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
+
+    def test_restart(self, verify_vm_running, vm):
+        assert "restart" in vm.action
+        vm.action.restart()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
```

Comment:
I think we might want to wait for the VM to go down first and then come back up.